pkgs/development/python-modules/connexion/default.nix

```
@@ -0,0 +1,70 @@
+{ buildPythonPackage
+, fetchFromGitHub
+, gevent
+, flake8
+, jsonschema
+, pytest
+, clickclick
+, pyyaml
+, openapi-spec-validator
+, inflection
+, requests
+, flask
+, mock
+, aiohttp
+, aiohttp-jinja2
+, ujson
+, six
+, pathlib
+, typing
+, swagger-ui-bundle
+, pytest-aiohttp
+, lib
+, pythonOlder
+, isPy3k}:
+
+buildPythonPackage rec {
+  pname = "connexion";
+  version = "2.1.0";
+
+  src = fetchFromGitHub {
+    owner = "zalando";
+    repo = pname;
+    rev = version;
+    sha256 = "1psbql502xa7g8q4m7mpqpdicyb2kffgq9wvrq9jr7ld22k2m282";
+  };
+
+  checkInputs = [ 
+    flake8
+    mock
+    pytest
+  ] ++ lib.optional isPy3k pytest-aiohttp;
+  propagatedBuildInputs = [
+    clickclick
+    flask
+    gevent
+    inflection
+    jsonschema
+    openapi-spec-validator
+    pyyaml
+    requests
+    six
+    swagger-ui-bundle
+  ]
+  ++ lib.optional (pythonOlder "3.4") pathlib
+  ++ lib.optional (pythonOlder "3.6") typing
+  ++ lib.optional isPy3k aiohttp
```

Forgot to mention `lib.optionals` which takes a list. This would make this look simpler
```suggestion
  ++ lib.optionals isPy3k [ aiohttp aiohttp-jinja2 json ]
```

I tend to use `optional` when it's singular and `optionals` when there's multiple.

---

Thanks for that tip! I've changed it to "optionals".